### PR TITLE
auth: log client for failed authentication

### PR DESCRIPTION
**Pull request by howardjohn**
Trying to debug auth errors, its pretty very hard without context

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: security/pkg/server/ca/server.go [hunk 1]
```diff
@@ -82,7 +92,6 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 	s.monitoring.CSR.Increment()
 	caller := s.authenticate(ctx)
 	if caller == nil {
-		serverCaLog.Warn("request authentication failure")
```
Comment (lei-tang) [inline security/pkg/server/ca/server.go:85]: Why this line of code is removed?

Comment (howardjohn) [reply security/pkg/server/ca/server.go:85]: we don't need to log it twice, authenticate already logs

Comment (lei-tang) [reply security/pkg/server/ca/server.go:85]: If caller == nil, the function will return (i.e., line 96: nil, status.Error(codes.Unauthenticated, "request authenticate failure")) and will not reach the new logging statement (i.e., line 348, serverCaLog.Warnf("Authentication failed for %v: %s", getConnectionAddress(ctx), errMsg)). 
Can you elaborate on logging it twice?

Comment (howardjohn) [reply security/pkg/server/ca/server.go:85]: line 348 happens prior to returning. Caller can only be nil if we have already reached l348

Comment (lei-tang) [reply security/pkg/server/ca/server.go:85]: Thanks for the clarification!

**Pull request merged**